Update quick start docs (#11140)
PyTorchLightning/pytorch-lightning: docs/source/starter/installation.rst

.. _installation:

############

> Installation

Reviewer: This isn't about installation. Maybe let's rename this Lightning Coverage

Author: it's about installation. Lightning coverage is just a small part to indicate whether users' env will be supported with PL or not.

############

> PyTorch Lightning is maintained and tested on Pytorch 3.7.0+ and PyTorch 1.7.0+.

Reviewer: This should be a more exhaustive matrix, similar to the readme.

Author: I can improve it a little more, but adding that exhaustive matrix might be an overkill. I don't think users might be interested in what kind of tests we cover in the docs. Plus we need to make sure that we update this particular doc every time a small change is made in the ci config. As far as I can tell, the coverage on README is a little outdated too.

Reviewer 2: If this is already in the README, I would not copy it here.

We will forget about this eventually. My suggestion is to remove it or link to the README.

Author: linked the README.

Install any supported version of PyTorch if you want from `PyTorch Installation Page <https://pytorch.org/get-started/locally/#start-locally>`_.
Now you can install using `pip <https://pypi.org/project/pytorch-lightning/>`_ using the following command:

.. code-block:: bash

    pip install pytorch-lightning

***********************
Installation with Conda
***********************

If you don't have conda installed, follow the `Conda Installation Guide <https://docs.conda.io/projects/conda/en/latest/user-guide/install>`_.
Lightning can be installed with `conda <https://anaconda.org/conda-forge/pytorch-lightning>`_ using the following command:

.. code-block:: bash

    conda install pytorch-lightning -c conda-forge

You can also use `Conda Environments <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html>`_:

.. code-block:: bash

    conda activate my_env
    conda install pytorch-lightning -c conda-forge

************************
Installation from Source
************************

Install future patch release from the source. Note that patch release contains only the bugfixes for the recent major release.

.. code-block:: bash

>     pip install git+https://github.com/PytorchLightning/pytorch-lightning.git@release/1.5.x --upgrade

Reviewer 3: ```suggestion
    pip install https://github.com/PyTorchLightning/pytorch-lightning/archive/refs/heads/release/1.5.x.zip
```
git is heavy

It is rigorously tested across multiple GPUs, TPUs, CPUs and IPUs. GPU tests run on two NVIDIA P100. TPU tests run on Google GKE TPUv2/3.
TPU py3.7 means we support Colab and Kaggle env. IPU tests run on MK1 IPU boxes.

*********************
Installation with Pip
*********************

Install any supported version of PyTorch if you want from `PyTorch Installation Page <https://pytorch.org/get-started/locally/#start-locally>`_.
Now you can install using `pip <https://pypi.org/project/pytorch-lightning/>`_ using the following command:

.. code-block:: bash

    pip install pytorch-lightning

***********************
Installation with Conda
***********************

If you don't have conda installed, follow the `Conda Installation Guide <https://docs.conda.io/projects/conda/en/latest/user-guide/install>`_.
Lightning can be installed with `conda <https://anaconda.org/conda-forge/pytorch-lightning>`_ using the following command:

.. code-block:: bash

    conda install pytorch-lightning -c conda-forge

You can also use `Conda Environments <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html>`_:

.. code-block:: bash

    conda activate my_env

>     conda install pytorch-lightning -c conda-forge

Reviewer 2: What's the benefit of mentioning this? This is not lightning-specific right?

Author: I think it's fine. If someone is not familiar with conda at all, then it's just a useful reference.

************************
Installation from Source
************************

Install future patch release from the source. Note that patch release contains only the bugfixes for the recent major release.

.. code-block:: bash

    pip install https://github.com/PyTorchLightning/pytorch-lightning/archive/refs/heads/release/1.5.x.zip

Install from the source. Note that it contains all the bugfixes and newly released features that

> are not published yet. Also newly added features might contain bugs so no gurantess.

Reviewer 2: No guarantees reads a bit aggresive to me

```suggestion
are not published yet. This is the bleeding edge so use it at your own discretion.
```

.. _installation:

############
Installation
############

--------------

******************
Lightning Coverage
******************

> PyTorch Lightning is maintained and tested on different Pytorch and PyTorch versions.

Reviewer 4: ```suggestion
PyTorch Lightning is maintained and tested on different PyTorch versions.
```

Author: ah! should be Python.. my fingers have learnt `Pytorch` word. 😅

Reviewer 4: Please help my eyes are bleeding. 
All the people spelling PyTorch wrong...

> PyTorch Lightning is maintained and tested on different Pytorch and PyTorch versions.

Author: ```suggestion
PyTorch Lightning is maintained and tested on different Python and PyTorch versions.
```
